### PR TITLE
Parametrize accepts sequence only and in python3 map cannot act as a list

### LINKED ISSUE #5856: Need python3 fixe for the below issue
```
==================================== ERRORS ====================================
______________ ERROR collecting tests/foreman/cli/test_setting.py ______________
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/pluggy/__init__.py:617: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/pluggy/__init__.py:222: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/pluggy/__init__.py:216: in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/_pytest/python.py:200: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/_pytest/python.py:378: in _genfunctions
    self.ihook.pytest_generate_tests(metafunc=metafunc)
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/pluggy/__init__.py:617: in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/pluggy/__init__.py:222: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/pluggy/__init__.py:216: in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/_pytest/python.py:125: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/_pytest/python.py:823: in parametrize
    if len(ids) != len(parameters):
E   TypeError: object of type 'map' has no len()
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
```

### PR DESCRIPTION
**Pull request by lpramuk**
```
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/_pytest/python.py:125: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
../../shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/lib/python3.6/site-packages/_pytest/python.py:823: in parametrize
    if len(ids) != len(parameters):
E   TypeError: object of type 'map' has no len()
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
```
Fixes #5856

### CODE REVIEW

**Comment by lpramuk** [comment]
Tested
```
$ /home/jenkins/shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/bin/py.test --collect-only -v tests/foreman/cli/test_setting.py
================================================= test session starts ==================================================
platform linux -- Python 3.6.4, pytest-3.4.2, py-1.5.2, pluggy-0.6.0 -- /home/jenkins/shiningpanda/jobs/375dbdea/virtualenvs/d41d8cd9/bin/python3.6
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jenkins/workspace/automation-6.3-tier1-rhel7, inifile:
plugins: xdist-1.22.2, services-1.2.1, mock-1.6.3, forked-0.2
collected 16 items                                                                                                     
2018-03-21 06:31:52 - conftest - DEBUG - Collected 16 test cases

<Module 'tests/foreman/cli/test_setting.py'>
  <UnitTestCase 'SettingTestCase'>
    <TestCaseFunction 'test_negative_discover_host_with_invalid_prefix'>
    <TestCaseFunction 'test_negative_update_email_reply_address'>
    <TestCaseFunction 'test_negative_update_email_subject_prefix'>
    <TestCaseFunction 'test_negative_update_hostname_with_empty_fact'>
    <TestCaseFunction 'test_positive_update_email_delivery_method_sendmail'>
    <TestCaseFunction 'test_positive_update_email_delivery_method_smtp'>
    <TestCaseFunction 'test_positive_update_email_reply_address'>
    <TestCaseFunction 'test_positive_update_email_subject_prefix'>
    <TestCaseFunction 'test_positive_update_hostname_default_facts'>
    <TestCaseFunction 'test_positive_update_hostname_default_prefix'>
    <TestCaseFunction 'test_positive_update_hostname_prefix_without_value'>
    <TestCaseFunction 'test_positive_update_login_page_footer_text'>
    <TestCaseFunction 'test_positive_update_login_page_footer_text_with_long_string'>
    <TestCaseFunction 'test_positive_update_login_page_footer_text_without_value'>
  <Function 'test_positive_update_send_welcome_email'>
  <Function 'test_negative_update_send_welcome_email[0]'>

================================================== 0 tests deselected ==================================================
============================================= no tests ran in 0.28 seconds =============================================
```

File: robottelo/datafactory.py [hunk 1]
```diff
@@ -146,7 +146,9 @@ def test_something(value):
     :param argvalues: to be passed to parametrize
     :return: dict
     """
-    return {'argvalues': argvalues, 'ids': map(str, range(len(argvalues)))}
+    return {
+        'argvalues': argvalues, 'ids': list(map(str, range(len(argvalues))))
```
Comment (abalakh) [inline robottelo/datafactory.py]: Why do we need that `map` here after all? Just to have indexes as strings? We could use something simpler and more descriptive like list generator, e.g.:
```python
'ids': [str(index) for index in range(len(argvalues))]
```

Comment (lpramuk) [reply robottelo/datafactory.py]: Good tip. It's useless to list(map([list]). I'll do it as you suggested.

Generally map is a *lot* faster then list comprehension, I know here we don't chase performance :-)
```
$ python3 -m timeit "map(str, range(1000))"
1000000 loops, best of 3: 0.385 usec per loop

$ python3 -m timeit "[str(index) for index in range(1000)]"
10000 loops, best of 3: 193 usec per loop
```

Comment (lpramuk) [reply robottelo/datafactory.py]: but after converting map to list performance is almost that slow like with list comprehension, heh
```
$ python3 -m timeit "list(map(str, range(1000)))"
10000 loops, best of 3: 152 usec per loop
```
152 vs. 193

Comment (abalakh) [reply robottelo/datafactory.py]: It's not 500 times faster, mapping is lazy, nothing was actually calculated until you tried to access elements or convert them to a list :)

Comment (lpramuk) [reply robottelo/datafactory.py]: @abalakh You seem to know a lot about snake guts ;-) Thx

**Comment by lpramuk** [comment]
Addressed @abalakh 's  comment.

**Pull request merged**